### PR TITLE
Storage - Load hashes only

### PR DESCRIPTION
**Pull request by tsvisabo**
# Test Plan:
One test is Included
Opened issue #138

### CODE REVIEW

File: common/storage/sql/sqlite3/storage.c [hunk 1]
```diff
@@ -26,6 +26,15 @@ typedef struct iota_transactions_pack {
   size_t num_loaded;
 } iota_transactions_pack;
 
+typedef struct iota_hashes_pack {
+  trit_array_p* hahses;
```
Comment (th0br0) [inline common/storage/sql/sqlite3/storage.c]: typo

File: common/storage/sql/sqlite3/storage.c [hunk 2]
```diff
@@ -140,6 +149,50 @@ retcode_t iota_stor_exist(const connection_t* const conn,
...
+                                       trit_array_p data_out[],
+                                       size_t max_num_hashes,
+                                       size_t* num_loaded) {
+  char statement[MAX_SELECT_STATEMENT_SIZE];
+  iota_transactions_select_hashes_statement(col_name, key, statement,
+                                            MAX_SELECT_STATEMENT_SIZE);
+
+  char* err_msg = 0;
+  iota_hashes_pack pack;
+  pack.hahses = data_out;
```
Comment (th0br0) [inline common/storage/sql/sqlite3/storage.c]: typo

File: common/storage/sql/sqlite3/storage.c [hunk 3]
```diff
@@ -140,6 +149,50 @@ retcode_t iota_stor_exist(const connection_t* const conn,
...
+                                       const trit_array_p key,
+                                       trit_array_p data_out[],
+                                       size_t max_num_hashes,
+                                       size_t* num_loaded) {
+  char statement[MAX_SELECT_STATEMENT_SIZE];
+  iota_transactions_select_hashes_statement(col_name, key, statement,
+                                            MAX_SELECT_STATEMENT_SIZE);
+
+  char* err_msg = 0;
+  iota_hashes_pack pack;
```
Comment (th0br0) [inline common/storage/sql/sqlite3/storage.c]: this should be passed into the method?

File: common/storage/sql/statements.c [hunk 4]
```diff
@@ -127,3 +127,27 @@ retcode_t iota_transactions_update_statement(const char *index_col,
                                              const iota_transaction_t tx,
                                              char statement[],
                                              size_t statement_cap) {}
+
+retcode_t iota_transactions_select_hashes_statement(const char *index_col,
+                                                    const trit_array_p key,
```
Comment (th0br0) [inline common/storage/sql/statements.c:132]: Let's use an `enum` for this kind of key?

File: common/storage/sql/statements.c [hunk 5]
```diff
@@ -127,3 +127,27 @@ retcode_t iota_transactions_update_statement(const char *index_col,
...
+    memcpy(key_str, key->trits, key->num_bytes);
+    key_str[key->num_bytes] = 0;
+    res = snprintf(statement, statement_cap, "SELECT %s FROM %s WHERE %s='%s'",
+                   COL_HASH, TRANSACTION_TABLE_NAME, index_col, key_str);
+  }
+
+  if (res < 0 || res == statement_cap) {
+    log_error(SQL_STATEMENTS_ID, "Failed in creating statement, statement: %s",
+              statement);
+    return RC_SQL_FAILED_WRITE_STATEMENT;
```
Comment (th0br0) [inline common/storage/sql/statements.c:151]: this is a `read` statement, not a `write` statement

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/storage/sql/sqlite3/tests/test_sqlite3.c [hunk 6]
```diff
@@ -73,6 +74,28 @@ void test_stored_transaction(void) {
   }
 }
 
+void test_stored_load_hashes_by_address(void) {
+  trit_array_p hashes[5];
+
+  for (int i = 0; i < 5; ++i) {
+    hashes[i] = trit_array_new(FLEX_TRIT_SIZE_243);
```
Comment (thibault-martinez) [inline common/storage/sql/sqlite3/tests/test_sqlite3.c]: Shouldn't it be just 243 here (not flex) ?
`trit_array_new` already takes care of the conversion.

File: common/storage/sql/sqlite3/storage.c (hunk 3 above)
Comment (tsvisabo) [reply common/storage/sql/sqlite3/storage.c]: yes, that would be better

File: common/storage/sql/statements.c (hunk 5 above)
Comment (tsvisabo) [reply common/storage/sql/statements.c:151]: Yes, but we are writing the read statement :)

File: common/storage/sql/sqlite3/tests/test_sqlite3.c (hunk 6 above)
Comment (tsvisabo) [reply common/storage/sql/sqlite3/tests/test_sqlite3.c]: That is correct, thanks

File: common/storage/sql/statements.c (hunk 4 above)
Comment (tsvisabo) [reply common/storage/sql/statements.c:132]: But the encoding can be different, so the enumToKeyTable (if that is what you have in mind) will need entry per column per encoding (1/3/4/5)

Comment (th0br0) [reply common/storage/sql/statements.c:132]: Nevermind, I was confusing index_col & key

**Review by th0br0 — APPROVED** [review]

**Pull request merged**